code/game/objects/items/devices/flashlight.dm

```diff
@@ -532,3 +532,48 @@
 	flashlight_power = 1
 	flags_1 = CONDUCT_1 | DROPDEL_1
 	actions_types = list()
+
+
+
+/obj/item/device/flashlight/directional
+	name = "old flashlight"
+	desc = "This poor thing has seen better days, it only emits light in a single direction"
+	icon_state = "oldlight"
+	dir = 4
+	brightness_on = 3
+	var/pointing
+
+/obj/item/device/flashlight/directional/update_brightness(mob/user = null)
+	..()
+	if(light)
+		if(user)
+			dir = user.dir //Someone is holding the light, so aim it where they are facing
+		light.directional = dir2angle(dir)
+		update_light()
+
+/obj/item/device/flashlight/directional/Initialize()
+	..()
+	START_PROCESSING(SSobj, src)
+
+/obj/item/device/flashlight/directional/Destroy()
+	STOP_PROCESSING(SSobj, src)
+	. = ..()
+
+/obj/item/device/flashlight/directional/process() //So that the light follows you when you turn without moving

```

Comment:
I'm not sure I am keen on this approach.

The flashlight should register as a signal handler for the mob and then be sent signals when the mob's dir is updated.

I'll have to review our code and see how we do mob dir updates, we might be doing it haphazardly

@tgstation/commit-access do we know off the top of our head if all direction changes for mobs are done through a proc or people just set the dir?

Reply:
I'm using Naksu's component example and seeing if that will just be the silver bullet for this problem.